trl/trainer/ppo_trainer.py

```diff
@@ -428,14 +431,28 @@ def step(
 
         with torch.no_grad():
             all_logprobs, _, values, masks = self.batched_forward_pass(self.model, queries, responses, model_inputs)
-            ref_logprobs, _, _, _ = self.batched_forward_pass(self.ref_model, queries, responses, model_inputs)
+            # for when the model is a peft model
+            if self.is_peft_model and hasattr(self.model.pretrained_model, "disable_adapter"):
+                with self.model.pretrained_model.disable_adapter():
+                    ref_logprobs, _, _, _ = self.batched_forward_pass(self.model, queries, responses, model_inputs)
+            elif self.is_peft_model and not hasattr(self.model.pretrained_model, "disable_adapter"):
+                raise ValueError(
+                    "You are using a `peft` version that does not support `disable_adapter`. Please update your `peft` version to the latest version."
+                )
+
+            else:
+                ref_logprobs, _, _, _ = self.batched_forward_pass(self.ref_model, queries, responses, model_inputs)
 
         timing["time/ppo/forward_pass"] = time.time() - t
 
         t = time.time()
         rewards, non_score_reward = self.compute_rewards(scores, all_logprobs, ref_logprobs, masks)
         timing["time/ppo/compute_rewards"] = time.time() - t
 
+        if self.is_peft_model:
+            self.model.train()
+            self.model.gradient_checkpointing_enable()
+
```

Comment:
Isn't that something we want to do outside of the trainer or at least make optional? otherwise GC is always turned on, no?

Reply:
It is recommended for users to enable GC when using `peft` to maximize efficiency, maybe we can have a check that warns users to turn on GC if `peft` is available

Reply:
let's just educate users on how to properly use this instead on the example